echo/README.md

```diff
@@ -1,5 +1,233 @@
 # Echo Example
 
-This example is designed to illustrate a few simple aspects of coding an actor module with wasmCloud. The first shows how to respond to simple HTTP requests, but this sample also shows how to use `serde` for JSON serialization.
+An actor is the smallest unit of deployable, portable compute within the wasmcloud ecosystem.
 
-This actor module responds to all incoming HTTP requests by returning a JSON object describing the inbound request.
+Actors are small WebAssembly modules that can handle messages delivered to them by the host runtime and can invoke functions on capability providers, provided they have been granted the appropriate privileges.
+
+Check out https://wasmcloud.dev/reference/host-runtime/actors/ for more.
+
+In this example our actor responds to all incoming HTTP requests by returning the HTTP request as a JSON object.
+
+We will have a two part tutorial, starting with running a precompiled actor stored in a remote repo, then we will write our own actor and swap it with the remote one. We will use a manifest file to run our actor declaratively via `wasmcloud`.
+
+In the second part, we will spin up a custom NATS server and a local docker registry to store our WebAssembly module. We'll use the wasmcloud shell, `wash`, a feature-rich cli tool that provides an interactive REPL environment to run our actor and interact with it.
+
+## Running an echo server
+
+1. Please install `wasmcloud` and `wash` binaries if you do not have them installed already. `wasmcloud` will provide the runtime environment for our actor while `wash` will be used for development later in the tutorial:
+
+    https://wasmcloud.dev/overview/installation/
```

Comment:
Same as above comment, we can use a markdown link to make this clickable directly.